src/main/webapp/app/exercises/programming/manage/grading/tasks/programming-exercise-task/programming-exercise-task.component.ts

```diff
@@ -75,4 +78,19 @@ export class ProgrammingExerciseTaskComponent implements OnInit {
     formatTestIndex(i: number, j: number): string {
         return `${i + 1}.${j + 1}`;
     }
+
+    private updateTestCaseVisibilityList() {
+        this.testCaseVisibilityList = Object.entries(Visibility).map(([name, value]) => {
+            let displayName = name;
+
+            if (this.isExamExercise && value == Visibility.AfterDueDate) {
```

Comment:
We should check for type, too. --> if (this.isExamExercise && value === Visibility.AfterDueDate) {